File: lib/csharp/microsoft.bot.builder.skills/Microsoft.Bot.Builder.Skills.Tests/Mocks/MockSkillTransport.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.Bot.Builder.Skills.Auth;
using Microsoft.Bot.Builder.Skills.Models.Manifest;
using Microsoft.Bot.Schema;

namespace Microsoft.Bot.Builder.Skills.Tests.Mocks
{
    public class MockSkillTransport : ISkillTransport
    {
        private Activity _activityForwarded;

		public Task CancelRemoteDialogsAsync(SkillManifest skillManifest, IServiceClientCredentials serviceClientCredentials, ITurnContext turnContext, CancellationToken cancellationToken = default)
		{
			return Task.CompletedTask;
		}

        public void Disconnect()
        {
        }

		public Task<Activity> ForwardToSkillAsync(SkillManifest skillManifest, IServiceClientCredentials serviceClientCredentials, ITurnContext dialogContext, Activity activity, Action<Activity> tokenRequestHandler = null, Action<Activity> fallbackHandler = null, CancellationToken cancellationToken = default)
		{
			_activityForwarded = activity;

            return Task.FromResult<Activity>(null);
        }

        public bool CheckIfSkillInvoked()
        {
            return _activityForwarded != null;
        }

        public void VerifyActivityForwardedCorrectly(Action<Activity> assertion)
        {
            assertion(_activityForwarded);
        }
    }
}

File: sdk/typescript/libraries/botbuilder-skills/test/mocks/mockSkillTransport.js
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

class MockSkillTransport {

    async cancelRemoteDialog(turnContext) {
        return Promise.resolve();
    }

    disconnect() { }

    async forwardToSkill(turnContext, activity, tokenRequestHandler) {
        this.activityForwarded = activity;
        return Promise.resolve(true);
    }

    checkIfSkillInvoked() {
        return this.activityForwarded !== undefined;
    }

    verifyActivityForwardedCorrectly(activityAssertion){
        activityAssertion(this.activityForwarded);
    }
}

exports.MockSkillTransport = MockSkillTransport;